x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportInternalInferModelAction.java

```diff
@@ -197,9 +206,8 @@ private void inferSingleDocAgainstAllocatedModel(
             request,
             ActionListener.wrap(r -> listener.onResponse(r.getResults()), e -> {
                 Throwable unwrapped = ExceptionsHelper.unwrapCause(e);
-                if (unwrapped instanceof ElasticsearchStatusException) {
-                    ElasticsearchStatusException ex = (ElasticsearchStatusException) unwrapped;
-                    if (ex.status().equals(RestStatus.TOO_MANY_REQUESTS)) {
+                if (unwrapped instanceof ElasticsearchException ex) {
+                    if (FAILURE_STATUSES.contains(ex.status())) {
```

Comment:
Looking at the changes in `FillMaskProcessor` it makes more sense for individual processors to decide which errors are exceptions and which are `WarningInferenceResults`. I think this logic and be removed as the processor and deployment manager know best what is an error

Reply:
@davidkyle this assumes that we catch and return a warning down at the task level. I can do that but its a larger refactor.